src/Symfony/Bridge/Monolog/Handler/ElasticsearchLogstashHandler.php

```diff
@@ -149,11 +149,17 @@ private function sendToElasticsearch(array $records): void
         $this->wait(false);
     }
 
+    /**
+     * @return never
```

Comment:
I'm not a fan of the added `never` types to sleep+wakeup. To me, this is motivated by the internal implementation, but the implementation shouldn't drive the contract. Since these methods are magic, they're not supposed to be called by any userland code directly. This more tailored type doesn't serve any purpose. Actually, it only prevents potential child classes from providing their own implementation if they'd like to. I don't have a use case in mind, but that's not a reason to clone the API here to me.

Reply:
I'm also a bit hesitant to add `never` as a return type, as it blocks the user from implementing any other behavior.

But in this case, I added them because all these sleep+wakeup calls are `never` to prevent insecure deserialization. I thought it would be a good idea to actually prevent user-code from loosing this requirement, as that could unknowingly make them vulnerable for this type of attacks.

Reply:
Sure and I understand that. That's on the edge of the open/closed principles, just a bit too much on the closed side :)

Reply:
I realized that the security argument doesn't stand: implementing __serialize still allows working around never; this adds to "open" side.